pkg/cmd/adm/register_member.go

```diff
@@ -120,66 +114,210 @@ func registerMemberCluster(ctx *extendedCommandContext, newCommand client.Comman
 		return nil
 	}
 
-	return validated.perform(ctx, newCommand)
+	return validated.perform(ctx)
+}
+
+func (v *registerMemberValidated) getSourceAndTargetClusters(sourceClusterType configuration.ClusterType) (clusterData, clusterData) {
+	if sourceClusterType == configuration.Member {
+		return v.memberClusterData, v.hostClusterData
+	}
+	return v.hostClusterData, v.memberClusterData
 }
 
-func runAddClusterScript(term ioutils.Terminal, newCommand client.CommandCreator, joiningClusterType configuration.ClusterType, hostKubeconfig, hostNs, memberKubeconfig, memberNs, nameSuffix string, useLetsEncrypt bool) error {
-	if !term.AskForConfirmation(ioutils.WithMessagef("register the %s cluster by creating a ToolchainCluster CR, a Secret and a new ServiceAccount resource?", joiningClusterType)) {
+// addCluster creates a secret and a ToolchainCluster resource on the `targetCluster`.
+// This ToolchainCluster CR stores a reference to the secret which contains the kubeconfig of the `sourceCluster`. Thus enables the `targetCluster` to interact with the `sourceCluster`.
+// - `targetCluster` is the cluster where we create the ToolchainCluster resource and the secret
+// - `sourceCluster` is the cluster referenced in the kubeconfig/ToolchainCluster of the `targetCluster`
+func (v *registerMemberValidated) addCluster(ctx *extendedCommandContext, sourceClusterType configuration.ClusterType) error {
+	if !ctx.AskForConfirmation(ioutils.WithMessagef("register the %s cluster by creating a ToolchainCluster CR, a Secret and a new ServiceAccount resource?", sourceClusterType)) {
 		return nil
 	}
```

Comment:
I would drop this question as it can be confusing for people who are not familiar with the architecture. You know, the command is called `register-member` but this question is asking if they are OK to register host :-) 
I would just print this instead:
```suggestion
	ctx.PrintContextSeparatorf("Ensuring connection from the %s cluster to the %s via a ToolchainCluster CR, a Secret, and a new ServiceAccount resource", sourceClusterType, sourceClusterType.TheOtherType())
```

Reply:
good point! let's drop it!